### ISSUE TITLE
Small bug with autocomplete suggestion menu in Listener

### ISSUE THREAD

**Issue by h0le** [factor/factor#2022]
I think this is not critical defect, but it's confusing a little. Autocomplete suggestion menu in Listener behaves inadequate sometimes. For example, when I type vocabulary name after `USE:` and then I clear whole line by `backspace`, autocomplete suggestion menu doesn't disappear, even after refreshing by `F2`. This is for `Win X86-64 2018-07-16` build of `0.98`.
![bug1](https://user-images.githubusercontent.com/4871017/42887896-5b223eda-8aea-11e8-881d-20418bcd39ca.png)

**Comment by erg** [comment]
How about hitting ESC?

I'm not sure exactly when to remove the popup, got a suggestion?

**Comment by mrjbq7** [comment]
Something like this could work, although I don't have time to properly test this, we can look into making it more robust later.

```diff
diff --git a/basis/ui/tools/listener/listener.factor b/basis/ui/tools/listener/ldiff --git a/basis/ui/tools/listener/listener.factor b/basis/ui/tools/listener/l
istener.factor
index 89fe0cd74a..e6169ee9bb 100644
--- a/basis/ui/tools/listener/listener.factor
+++ b/basis/ui/tools/listener/listener.factor
@@ -76,6 +76,17 @@ M: vocab-word-completion (word-at-caret)
         dup <word-model> >>word-model
         dup model>> <history> >>history ;
 
+: ?check-popup ( interactor -- interactor )
+    dup popup>> [
+        gadget-child dup completion-popup? [
+            completion-mode>> over completion-mode =
+            [ dup popup>> hide-glass ] unless
+        ] [ drop ] if
+    ] when* ;
+
+M: interactor user-input*
+    ?check-popup call-next-method ;
+
 M: interactor graft*
     [ call-next-method ] [ dup word-model>> add-connection ] bi ;
 
@@ -376,7 +387,7 @@ M: interactor stream-read-quot ( stream -- quot/f )
 M: interactor handle-gesture
     {
         { [ over key-gesture? not ] [ call-next-method ] }
-        { [ dup popup>> ] [ { [ pass-to-popup ] [ call-next-method ] } 2&& ] }
+        { [ dup popup>> ] [ ?check-popup { [ pass-to-popup ] [ call-next-method ] } 2&& ] }
         {
             [ dup token-model>> value>> ]
```

**Comment by mrjbq7** [comment]
P.S., @erg ``Ctrl-K`` doesn't delete from cursor to end of line for some reason, it deletes everything in the editor...

(I have this in my ``.factor-rc``):

```factor
USING: accessors assocs kernel sequences sets ui.commands
ui.gadgets.editors ui.gestures ui.tools.listener ;

"multiline" multiline-editor get-command-at [
    {
        { T{ key-down f { C+ } "k" } delete-to-end-of-line }
        { T{ key-down f { C+ } "a" } start-of-line }
        { T{ key-down f { C+ } "e" } end-of-line }
    } append members
] change-commands drop multiline-editor update-gestures
```

**Comment by mrjbq7** [comment]
Fixed in 972fa0fbab68cebe5f6ed2ed0bbb3d4996190e4f.
